graphql-dgs-spring-webflux-autoconfigure/src/main/kotlin/com/netflix/graphql/dgs/webflux/autoconfiguration/DgsWebFluxAutoConfiguration.kt

```diff
@@ -0,0 +1,194 @@
+        val graphQlHandler = GraphQlHandler(dgsQueryExecutor)
+
+        return RouterFunctions.route()
+            .POST(
+                configProps.path, accept(MediaType.APPLICATION_JSON, MediaType.valueOf("application/graphql")),
+                graphQlHandler::graphql
+            ).build()
+    }
+
+    @Bean
+    @ConditionalOnProperty(name = ["dgs.graphql.schema-json.enabled"], havingValue = "true", matchIfMissing = true)
+    open fun schemaRouter(schemaProvider: DgsSchemaProvider): RouterFunction<ServerResponse> {
+        return RouterFunctions.route()
+            .GET(
+                configProps.schemaJson.path
+            ) {
+                val graphQLSchema: GraphQLSchema = schemaProvider.schema()
+                val graphQL = GraphQL.newGraphQL(graphQLSchema).build()
+
+                val executionInput: ExecutionInput =
+                    ExecutionInput.newExecutionInput().query(IntrospectionQuery.INTROSPECTION_QUERY)
+                        .build()
+                val execute = graphQL.executeAsync(executionInput)
+
+                return@GET Mono.fromCompletionStage(execute)
+                    .map { it.toSpecification() }
+                    .flatMap { ok().json().bodyValue(it) }
+            }.build()
+    }
+
+    class GraphQlHandler(private val dgsQueryExecutor: DgsReactiveQueryExecutor) {
+        val logger: Logger = LoggerFactory.getLogger(GraphQlHandler::class.java)
+        val mapper = jacksonObjectMapper()
+
+        fun graphql(request: ServerRequest): Mono<ServerResponse> {
+            val executionResult: Mono<ExecutionResult> =
+
+                request.bodyToMono(String::class.java)
+                    .map {
+                        if ("application/graphql" == request.headers().firstHeader("Content-Type")) {
+                            QueryInput(it)
+                        } else {
+                            val readValue = mapper.readValue<Map<String, Any>>(it)
+                            QueryInput(
+                                readValue["query"] as String,
+                                readValue.getOrDefault("variables", emptyMap<String, Any>()) as Map<String, Any>,
+                                readValue.getOrDefault("extensions", emptyMap<String, Any>()) as Map<String, Any>
+                            )
+                        }
+                    }
+                    .flatMap { queryInput ->
+                        logger.debug("Parsed variables: {}", queryInput.queryVariables)
+
+                        dgsQueryExecutor.execute(
+                            queryInput.query,
+                            queryInput.queryVariables,
+                            queryInput.extensions,
+                            request.headers().asHttpHeaders(),
+                            "",
+                            request
+                        )
+                    }.subscribeOn(Schedulers.parallel())
```

Comment:
Can you explain, please. Why do we need here an parallel scheduler? What the benefit of using it?
I can't find any explanation in source code(

Reply:
I'm also having trouble understanding the value of `.subscribeOn(Schedulers.parallel())`. Also note that this _Scheduler_ is not a _TaskScheduler_ managed by Spring/Spring Boot, it will not be able to leverage the _delegating task executors_ that different Spring libraries might need. For example, in Spring Security there is the `DelegatingSecurityContextAsyncTaskExecutor` that makes sure a _SecurityContext_ is available.

Reply:
That shouldn't have been there, it's a leftover of something I was trying.